zmon_worker_monitor/builtins/plugins/scalyr.py

```diff
@@ -80,20 +81,29 @@ def facets(self, filter, field, max_count=5, minutes=30, prio='low'):
         j = r.json()
         return j
 
-    def timeseries(self, filter, function='count', minutes=30, buckets=1, prio='low'):
-
+    def timeseries(self, filter, function='count', minutes=30, buckets=1, prio='low', align=0):
+        startTime = str(minutes) + 'm'
+        endTime = None
+        if align != 0:
+            cur_time = int(time.time())  # this assumes the worker is running with UTC time
+            end = cur_time - (cur_time % align)
```

Comment:
I am really surprised the formula is easy as this, but it works.
Don't leap seconds heavily disrupt it? https://en.wikipedia.org/wiki/Leap_second

Reply:
That will eventually fail once when the check  is made in the leap second ... failing in the sense that it returns a float not ending with `.0`. You get a float back in any case, so turning to an `int()` would be a good help anyway